### PR TITLE
Enhanced Dhall fixer support

### LINKED ISSUE #2820: Support the Dhall configuration language
**Name:** Dhall
**URL:** https://dhall-lang.org/

Dhall is a configuration language provides a non-Turing-complete, strongly typed way to configure programs or generate configuration files.
The `dhall` interpreter program provides a `lint` and a `format` subcommand, which could respectively become a Linter and a Fixer.

### PR DESCRIPTION
**Pull request by toastal**
Should fix #2820 since it's just asking about `format` and `lint`, but there's room to also do the language server, but one thing at a time. I also included `freeze` as a fixer as well to get sha256 for remote resources which is handy.

There's support for adding options to the main `dhail` executable such as `--ascii` (something `dhall-vim` doesn't support) as well as flags for `freeze` and `--all` if you'd really want that.

- - -

[PR #3310](https://github.com/dense-analysis/ale/pull/3310) somewhat addresses Dhall support (only `dhall format`), but the author forgot to search for the appropriate referencing the linked issue, #2820, or helping out with this already-open PR (which has more features), and was merged in before this PR got any attention :disappointed:. That PR, #3310, doesn't provide any configuration options (I know some people will want the the `--ascii` flag for sure), and doesn't include the other fixer-related tools, `dhall-lint` or `dhall-freeze`. This PR, #3261, should replace it.

However, that PR#3310 did appropriately try to find the file as a Stack binary as well.

### CODE REVIEW

**Comment by toastal** [comment]
I could use a little assistance here.  I ran `./run-tests -v -q --linters-only` and just like the failed [CI job](https://travis-ci.com/github/dense-analysis/ale/jobs/367658156), I'm getting 

```
...
========================================

Check Python code

========================================

.....

----------------------------------------------------------------------

Ran 5 tests in 0.001s

OK

Something went wrong!

The command "./run-tests -v $OPTIONS" exited with 1.

Done. Your build exited with 1.
``` 
This output isn't very helpful as it all looks okay.

**Comment by toastal** [comment]
@w0rp tests are now passing; awaiting review.

File: autoload/ale/fix/registry.vim [hunk 1]
```diff
@@ -375,11 +390,6 @@ let s:default_registry = {
 \       'suggested_filetypes': ['html', 'htmldjango'],
 \       'description': 'Fix HTML files with html-beautify.',
 \   },
-\   'dhall': {
```
Comment (w0rp) [inline autoload/ale/fix/registry.vim:378]: Add an alias to `dhall-format` for this, so anyone who is currently using the fixer under this name can keep using it.

**Review by w0rp — CHANGES_REQUESTED** [review]

**Comment by toastal** [comment]
Alias was created @w0rp

**Comment by ldesgoui** [comment]
Been using this for a few days, works great, thanks a lot for taking the time to contribute

**Comment by ldesgoui** [comment]
This PR isn't stale as it's ready

**Review by hsanson — APPROVED** [review]
Looks good and indeed more feature-full.

**Pull request merged**